app/src/main/java/Bamboo/model/Game.java

```diff
@@ -47,7 +47,8 @@ public void placeNextAt(Vector v)
         }
         if(grid.isFinished(currentPlayer.getColor()))
         {
-            view.endGame(currentPlayer);
+            try {Thread.sleep(2000); } catch (Exception exception){}
+            view.endGame(getCurrentPlayer());
```

Comment:
This does not work because the turn is toggled before this check happens. I think if we move this part of the method up it should work.

Reply:
Realised I missed a line here, so I just made a method that takes the non-currentPlayer instead, although I think your solution is more elegant